src/Raven.Server/Documents/Indexes/Sharding/Persistence/Corax/ShardedCoraxIndexReadOperation.cs

```diff
@@ -51,8 +52,9 @@ public sealed class ShardedCoraxIndexReadOperation : CoraxIndexReadOperation
     private ShardedQueryResultDocument AddOrderByFields(Document queryResult, IndexQueryServerSide query, ref EntryTermsReader reader, OrderMetadata[] orderByFields)
     {
         var result = ShardedQueryResultDocument.From(queryResult);
+        var currentCoraxOrderIndex = 0;
 
-        for (int i = 0; i < query.Metadata.OrderBy.Length; i++)
+        for (int i = 0; i < query.Metadata.OrderBy.Length && currentCoraxOrderIndex < orderByFields.Length; i++)
```

Comment:
Under what scenario would they be different lengths?

Reply:
With changes made in this PR, we won't create `OrderMetadata` for fields with zero terms in `CoraxQueryBuilder.GetSortMetadata`. This means it's possible to have less order by fields from index than from query. E.g. query I added in the test, but with additional `.ThenBy()` using a field with non-zero terms executed on a sharded database would have two order by fields from query, but only one from index.

Reply:
This requires the comment in the code

Reply:
Added

Reply:
This is applicable to sharded database, although I don't see a test case which is using a sharded database. Can you add it?

Reply:
I want elaborate a little bit more about this, and generally about skipping order clauses in general. This is optimization for cases when e.g. tree has no terms and gives us possibility to not wrap our query in extra SortingMatch/MultiSortingMatch and avoid expensive memoization. General idea was "if tree has no terms (so matched documents by inner query contain no terms to sort) we can return order from inner because sorting will just  be NOOP.

Reply:
Added test case for sharding